CONTRIBUTING.md

```diff
@@ -0,0 +1,93 @@
+# How to contribute
+
+We welcome all contributions to this project. Be it bug reports or feature requests via the [Issues tab](https://github.com/pygfx/shadertoy/issues) or code contributions via [Pull Requests](https://github.com/pygfx/shadertoy/pulls).
+
+
+## Submitting a pull request 
+
+Find an open issue 
+`git` and basic git knowledge is required.
+
+1. Fork the [repository](https://github.com/pygfx/shadertoy) using the "Fork" button.
+2. Clone the repository locally and set the upstream remote to the original repository.
+```bash
+$ git clone https://github.com/<your-github-handle>/shadertoy.git
+$ cd shadertoy
+$ git remote add upstream https://github.com/pygfx/shadertoy.git
+```
+3. Create a new branch for your changes. Make sure to branch off from newest `main` branch.
+```bash
+$ git checkout main
+$ git getch upstream
+$ git pull 
+$ git checkout -b <your-change>
+```
+4. Ensure that your environment is set up for development. See [Developers](#developers) for more information
+5. Make your changes and commit them with a meaningful commit message.
+```bash
+$ git add .
+$ git commit -m "Add my-change"
+```
+6. Push your changes to your fork.
+```bash
+$ git push -u origin <your-change>
+```
+7. Once your changes are ready, ensure that your changes are formatted and pass tests. See [Testing](#testing) for details.
+Go to your fork on GitHub and click the "Compare & pull request" button. Link any connected issues in the description. It is okay to open a draft pull request should your work be not yet complete. 
+You might be asked to make changes to your pull request. Additional commits you push to your branch will show up on the pull request. After your pull request is approved, it will be merged into the main branch.
```

Comment:
```suggestion
7. Once your changes are ready, ensure that your changes are formatted and pass tests. See [Testing](#testing) for details.

Go to your fork on GitHub and click the "Compare & pull request" button. Link any connected issues in the description. It is okay to open a draft pull request should your work be not yet complete.

You might be asked to make changes to your pull request. Additional commits you push to your branch will show up on the pull request.

After your pull request is approved, it will be merged into the main branch.
```